src/transformers/models/colpali/modular_colpali.py

```diff
@@ -0,0 +1,771 @@
+    is_flash_attn_2_available,
+    logging,
+    replace_return_docstrings,
+)
+
+
+if is_flash_attn_2_available():
+    from flash_attn.bert_padding import index_first_axis, pad_input, unpad_input  # noqa
+
+
+logger = logging.get_logger(__name__)
+
+
+class ColPaliConfig(PaliGemmaConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`ColPaliForRetrieval`]. It is used to instantiate an
+    ColPaliForRetrieval according to the specified arguments, defining the model architecture.
+
+    The ColPali config is very similar to [`PaligemmaConfig`], but with an extra attribute defining the embedding dimension.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        vision_config (`PaliGemmaVisionConfig`,  *optional*):
+            Custom vision config or dict
+        text_config (`Union[AutoConfig, dict]`, *optional*):
+            The config object of the text backbone. Can be any of `LlamaConfig` or `MistralConfig`.
+        ignore_index (`int`, *optional*, defaults to -100):
+            The ignore index for the loss function.
+        image_token_index (`int`, *optional*, defaults to 256000):
+            The image token index to encode the image prompt.
+        vocab_size (`int`, *optional*, defaults to 257152):
+            Vocabulary size of the PaliGemmamodel. Defines the number of different tokens that can be represented by the
+            `inputs_ids` passed when calling [`~PaliGemmaForConditionalGeneration`]
+        projection_dim (`int`, *optional*, defaults to 2048):
+            Dimension of the multimodal projection space.
+        hidden_size (`int`, *optional*, defaults to 2048):
+            Dimension of the hidden layer of the Language model.
+        embedding_dim (`int`, *optional*, defaults to 128):
+            Dimension of the multi-vector embeddings produced by the model.
+
+    Example:
+
+    ```python
+    from transformers.models.colpali import ColPaliConfig, ColPaliForRetrieval
+
+    config = ColPaliConfig()
+    model = ColPaliForRetrieval(config)
+    ```
+    """
+
+    def __init__(
+        self,
+        vision_config=None,
+        text_config=None,
+        ignore_index=-100,
+        image_token_index=256000,
+        vocab_size=257152,
+        projection_dim=2048,
+        hidden_size=2048,
+        embedding_dim: int = 128,
+        **kwargs,
+    ):
+        super().__init__(
+            vision_config=vision_config,
+            text_config=text_config,
+            ignore_index=ignore_index,
+            image_token_index=image_token_index,
+            vocab_size=vocab_size,
+            projection_dim=projection_dim,
+            hidden_size=hidden_size,
+            **kwargs,
+        )
+        self.model_type = "colpali"
+        self.is_composition = False
+        self.embedding_dim = embedding_dim
+
+
```

Comment:
```suggestion
    def ignore_index(self):
        raise AttributeError("Not needed for ColPali")
```
this is how you "remove" properties / setters!